### PR TITLE
Update shop filter

### PR DESCRIPTION
**Pull request by tyrosine1153**
### Description

- Edit ShopState request 
  - add itemIds param to improve shop buy filter
  - edit ItemProducts caching dict to only list
- Add Shop Sort - EquipmentLevel, OptionCount, UnitPrice

\* refer commits

### Related Links

[23.2Q 샵 개선 - 상점정렬, 필터 수정](https://app.asana.com/0/958521740385861/1204439171665346/f)

### CODE REVIEW

File: nekoyume/Assets/_Scripts/UI/Shop/BuyView.cs [hunk 1]
```diff
@@ -70,6 +69,8 @@ public enum BuyMode
         [SerializeField]
         private GameObject loading;
 
+        private int _loadingCount;
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/UI/Shop/BuyView.cs:72]: `_loadingCount` is used for what purpose?

File: nekoyume/Assets/_Scripts/State/ReactiveShopState.cs [hunk 2]
```diff
@@ -28,173 +26,102 @@ public static class ReactiveShopState
...
             var itemSubType = filter.ToItemSubType();
             var statType = filter.ToItemStatType();
-            var (products, totalCount) = await Game.Game.instance.MarketServiceClient.GetBuyProducts(itemSubType, offset, limit, orderType, statType);
-            // Debug.Log($"[RequestBuyProductsAsync] : {itemSubType} / {filter} / {orderType} / {offset} / {limit} / {statType} / MAX:{totalCount}");
+            var offset = reset ? 0 : CachedBuyItemProducts.Count;
+            var (products, totalCount) =
+                await Game.Game.instance.MarketServiceClient.GetBuyProducts(itemSubType, offset, limit, orderType, statType, itemIds);
 
-            await foreach (var product in products)
+            if (offset == 0)
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/State/ReactiveShopState.cs:74]:
```suggestion
            if (reset)
```

File: nekoyume/Assets/_Scripts/State/ReactiveShopState.cs [hunk 3]
```diff
@@ -28,173 +26,102 @@ public static class ReactiveShopState
...
                 return;
             }
 
-            if (!CachedBuyFungibleAssetProducts[orderType].ContainsKey(filter))
+            var ticker = filter == ItemSubTypeFilter.RuneStone ? "RUNE" : "SOULSTONE";
+            var offset = reset ? 0 : CachedBuyFungibleAssetProducts.Count;
+            var (fungibleAssets, totalCount) =
+                await Game.Game.instance.MarketServiceClient.GetBuyFungibleAssetProducts(ticker, offset, limit, orderType, itemIds);
+
+            if (offset == 0)
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/State/ReactiveShopState.cs:109]:
```suggestion
            if (reset)
```

File: nekoyume/Assets/_Scripts/UI/Shop/BuyView.cs [hunk 4]
```diff
@@ -337,45 +347,79 @@ private async void OnUpdateAscending(bool isAscending)
...
+                : _selectedSortFilter.Value.ToMarketOrderType(_isAscending.Value);
+            var filteredItemIds = GetFilteredItemIds(filter);
+
+            _loadingCount++;
+            loading.SetActive(_loadingCount > 0);
             if (filter is ItemSubTypeFilter.RuneStone or ItemSubTypeFilter.PetSoulStone)
             {
-                ReactiveShopState.SetBuyFungibleAssets(orderType);
+                await ReactiveShopState.RequestBuyFungibleAssetsAsync(
+                    filter, orderType, limit * 5, reset, filteredItemIds);
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/UI/Shop/BuyView.cs:414]:
```suggestion
                    filter, orderType, limit * 15, reset, filteredItemIds);
```

**Review by hskim881028 — CHANGES_REQUESTED** [review]

**Review by sonohoshi — APPROVED** [review]

**Review by hskim881028 — APPROVED** [review]

**Review by sonohoshi — APPROVED** [review]

**Pull request merged**